app/controllers/application_controller.rb

```diff
@@ -18,8 +20,64 @@ def set_checkout_redirect
     end
   end
 
+  def enable_embedded_styles
+    session[:embedded_shopfront] = true
+    render json: {}, status: 200
```

Comment:
This bit doesn't effect the security headers, it's only used to switch the alternative CSS layout on and off.